mod/item.php

```diff
@@ -94,30 +76,97 @@ function item_post(App $a) {
+}
+
+function item_edit(int $uid, array $request, bool $preview, string $return_path)
+{
+	$post = Post::selectFirst(Item::ITEM_FIELDLIST, ['id' => $request['post_id'], 'uid' => $uid]);
+	if (!DBA::isResult($post)) {
+		DI::sysmsg()->addNotice(DI::l10n()->t('Unable to locate original post.'));
+		if ($return_path) {
+			DI::baseUrl()->redirect($return_path);
+		}
+		throw new HTTPException\NotFoundException(DI::l10n()->t('Unable to locate original post.'));
+	}
+
+	$post['edit'] = $post;
+	$post['file'] = Post\Category::getTextByURIId($post['uri-id'], $post['uid']);	
+
+	$post = item_process($post, $request, $preview, $return_path);
+
+	$fields = [
+		'title'    => $post['title'],
+		'body'     => $post['body'],
+		'attach'   => $post['attach'],
+		'file'     => $post['file'],
+		'location' => $post['location'],
+		'coord'    => $post['coord'],
+		'edited'   => DateTimeFormat::utcNow(),
+		'changed'  => DateTimeFormat::utcNow()
+	];
+
+	$fields['body'] = Item::setHashtags($fields['body']);
+
+	$quote_uri_id = Item::getQuoteUriId($fields['body'], $post['uid']);
+	if (!empty($quote_uri_id)) {
+		$fields['quote-uri-id'] = $quote_uri_id;
+		$fields['body']         = BBCode::removeSharedData($post['body']);
+	}
+
+	Item::update($fields, ['id' => $post['id']]);
+	Item::updateDisplayCache($post['uri-id']);
 
-	$parent_item = null;
-	$toplevel_item = null;
-	$toplevel_item_id = 0;
-	$toplevel_user_id = null;
+	if ($return_path) {
+		DI::baseUrl()->redirect($return_path);
+	}
 
-	$objecttype = null;
-	$profile_uid = DI::userSession()->getLocalUserId();
-	$posttype = ($_REQUEST['post_type'] ?? '') ?: Item::PT_ARTICLE;
+	throw new HTTPException\OKException(DI::l10n()->t('Post updated.'));
+}
 
-	if ($parent_item_id || $thr_parent_uri) {
-		if ($parent_item_id) {
-			$parent_item = Post::selectFirst(Item::ITEM_FIELDLIST, ['id' => $parent_item_id]);
-		} elseif ($thr_parent_uri) {
-			$parent_item = Post::selectFirst(Item::ITEM_FIELDLIST, ['uri' => $thr_parent_uri, 'uid' => $profile_uid]);
+function item_insert(int $uid, array $request, bool $preview, string $return_path)
+{
+	$emailcc = trim($request['emailcc']  ?? '');
+
+	$post = ['uid' => $uid];
+	$post = DI::contentItem()->initializePost($post);
+
+	$post['edit']      = null;
+	$post['post-type'] = $request['post_type'] ?? '';
+	$post['wall']      = $request['wall'] ?? true;
+	$post['parent']    = intval($request['parent'] ?? 0);
+	$post['pubmail']   = $request['pubmail_enable'] ?? false;
+	$post['created']   = $request['created_at'] ?? DateTimeFormat::utcNow();
+	$post['edited']    = $post['changed'] = $post['commented'] = $post['created'];
+	$post['app']       = '';
+	$post['inform']    = '';
+	$post['postopts']  = '';
+	$post['file']      = '';
+
+	if ($post['parent']) {
+		if ($post['parent']) {
+			$parent_item = Post::selectFirst(Item::ITEM_FIELDLIST, ['id' => $post['parent']]);
 		}
 
 		// if this isn't the top-level parent of the conversation, find it
 		if (DBA::isResult($parent_item)) {
```

Comment:
DBA::isResult is superfluous with “selectFirst”
```suggestion
		if ($parent_item) {
```